client/src/utils/serverRequest.js

```diff
@@ -0,0 +1,201 @@
+import axios from 'axios';
+import { checkItem } from '../utils/arrayFunction';
+
+export const handleFetchProducts = async (url) => {
+   const response = await axios({ method: 'GET', url: url });
+
+   if (response.status === 200 || response.status === 201) {
+      return response;
+   } else {
+      throw new Error('Failed to fetch products');
+   }
+};
+
+export const handleAddToCartItem = async ({
+   state,
+   dispatch,
+   product,
+   navigate,
+   notify,
+}) => {
+   if (checkItem(state.cart, product).length === 0) {
+      notify(`${product.name} Adding to the Cart`);
+      try {
+         const { status } = await axios({
+            method: 'POST',
+            url: `https://driftkart-backend.duhbhavesh.repl.co/cart`,
+            data: {
+               _id: product.id,
+               quantity: 1,
+            },
+         });
+
+         if (status === 200 || status === 201) {
+            dispatch({ type: 'ADD_CART_ITEM', payload: product });
+         }
+      } catch (error) {
+         console.log(error);
+      }
+   } else {
+      navigate('/cart');
+   }
+};
+
+export const handleAddRemoveWishlistItem = async ({
+   state,

```

Comment:
i liked how you separated the api calls as a utilily function....will also make similar changes to my project!

Reply:
Thank you!